### ISSUE TITLE
Snapshot: cannot delete incorrect snapshot created through HTTP API

### ISSUE THREAD

**Issue by gjed** [grafana/grafana#51189]
**What happened**:
Cannot delete an incorrect snapshot created with the Snapshot API
From API:
```
{
  "message": "Error while checking permissions for snapshot",
  "traceID": "00000000000000000000000000000000"
}
```
Logs:
```
logger=context traceID=00000000000000000000000000000000 userId=1 orgId=1 uname=grafana-admin t=2022-06-21T18:37:39.585177279+02:00 level=error msg="Error while checking permissions for snapshot" error="Unique identifier needed to be able to get a dashboard" remote_addr=10.3.3.10 traceID=00000000000000000000000000000000
logger=context traceID=00000000000000000000000000000000 userId=1 orgId=1 uname=grafana-admin t=2022-06-21T18:37:39.585217193+02:00 level=error msg="Request Completed" method=DELETE path=/api/snapshots/l2OeOHc6uqaWQmRXK0mz6SfbF6ePpqxS status=500 remote_addr=10.3.3.10 time_ms=23 duration=23.613911ms size=104 referer= traceID=00000000000000000000000000000000
```
From GUI:

![Screenshot from 2022-06-21 18-36-16](https://user-images.githubusercontent.com/12700875/174852647-1063715b-e90b-40c9-9805-394fc49e201c.png)

**What you expected to happen**:
Either of the following:

- being able to delete the snapshot with http api
- being able to delete the snapshot from the GUI

**How to reproduce it (as minimally and precisely as possible)**:
On a fresh grafana instance, call the api with the following payload:
```
{
      "dashboard": {},
      "expires": 3600
}
```

**Anything else we need to know?**:
Can be delete from the db:
```
delete from dashboard_snapshot ;
```

**Environment**:
- Grafana version: 9.0.0 b5c56f6371 (docker)
- Data source type & version: /
- OS Grafana is installed on: (Docker)
- User OS & Browser: Debian 11, Chrome
- Grafana plugins: /
- Others: /

**Comment by dprokop** [comment]
I can confirm this is reproducible in the current version. I've tracked this down to RBAC checking permissions for snapshot dashboard over here: https://github.com/grafana/grafana/blob/main/pkg/api/dashboard_snapshot.go#L329. Since the dashboard ID 0 cannot be loaded here https://github.com/grafana/grafana/blob/main/pkg/services/guardian/accesscontrol_guardian.go#L72 (errors with `Unique identifier needed to be able to get a dashboard`), the API call yields permission check error. 

@grafana/grafana-authnz-team any suggestions on how we could tackle this problem?

**Comment by gamab** [comment]
@dprokop thanks for pinging us. I'll look into this.

**Comment by gamab** [comment]
Fix has been merged in main, backported to 9.1.x and 9.0.x. It will be available on next release (9.0.9, 9.1.3).
Thanks for reporting :slightly_smiling_face:
